### DISCUSSION TITLE
I would like to draw a shape - i.e. a polygon - but have it stay as a fixed size regardless of zoom level.

### DISCUSSION THREAD

**Discussion by lukemurray** [visgl/deck.gl#6574] [Q&A]
As the title - I would like to draw a shape - i.e. a polygon - but have it stay as a fixed size regardless of zoom level.

Think the IconLayer but with a complex shape. So at a certain point of the map draw a rect (for example) that is 10px x 10px regardless of zoom.

Is there any way to achieve this currently?

Thanks

**Comment by TextTL** [comment]
I have to scale using scale factor = desired zoom / (2**zoom level) to maintain the fixed size at certain zoom level.

**Comment by ibgreen** [comment]
One approach could be to use the deck.gl view system, overlaying a second view that is connected to a different viewState that you don't change when the user zooms. You would use the layerFilter prop to determine which view renders each layer.
